File: src/models/user_model.ts
import mongoose from 'mongoose';

export interface IUser {
  phoneNumber: string;
  name?: string;
  address?: string;
  email?: string;
  createdAt?: Date;
  token?: string; 
}

const userSchema = new mongoose.Schema<IUser>({
  phoneNumber: { type: String, required: true, unique: true },
  name: { type: String },
  address: { type: String },
  email: { type: String },
  createdAt: { type: Date },
  token: { type: String }, 
}, { timestamps: true });

export const UserModel = mongoose.model<IUser>('User', userSchema, 'users');

File: lib/models/user_model.dart
import 'package:json_annotation/json_annotation.dart';

part 'user_model.g.dart';

@JsonSerializable()
class User {
  String? name;
  String? phoneNumber;
  String? email;
  String? address;
  DateTime? createdAt;

  User({
    this.name,
    this.phoneNumber,
    this.email,
    this.address,
    this.createdAt,
  });

  factory User.fromJson(Map<String, dynamic> json) => _$UserFromJson(json);
  Map<String, dynamic> toJson() => _$UserToJson(this);
}